internal/cache/cache.go

```diff
@@ -0,0 +1,57 @@
+package cache
+
+import (
+	"github.com/Code-Hex/go-generics-cache/policy/lru"
+	"time"
+)
+
+type Cache[KeyT comparable, ValueT any] struct {
```

Comment:
A cache layer that could be used in getting separate entities from DB using queries or data loaders